provisioner/src/lib.rs

```diff
@@ -565,6 +573,31 @@ impl Provisioner for ShuttleProvisioner {
         Ok(Response::new(reply))
     }
 
+    #[tracing::instrument(skip(self))]
+    async fn dump_database(
+        &self,
+        request: Request<DatabaseDumpRequest>,
+    ) -> Result<Response<DatabaseDumpResponse>, Status> {
+        request.verify(Scope::Resources)?;
+        let claim = request.get_claim()?;
+        let request = request.into_inner();
+        if !ProjectName::is_valid(&request.project_name) {
+            return Err(Status::invalid_argument("invalid project name"));
+        }
+        self.verify_ownership(&claim, &request.project_name).await?;
+
+        let database_name = format!("db-{}", request.project_name);
+        let output = tokio::process::Command::new("pg_dump")
+            .arg(format!("{}{}", self.shared_pg_uri_stripped, database_name))
```

Comment:
Why do we first strip the database name, then add it back here?

Reply:
The incoming name from url is 'postgres', but here we need to change it to the name of the db to dump.